scripts/tests/provisioning/test.expect

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/expect -f
+#
+# Start a NILRT VM and instrument installing and running the provisioning
+# ptests. This script assumes that there is a running HTTP server at the ipv4
+# $vm_host_ip, on port 8990.
+# 
+# Expected arguments:
+#   1) path to the run-VM script
+#   2+) ipk feed paths (optional)
+log_file -noappend provisioning.log
+
+set ipk_feeds [lrange $argv 1 end]
+# The SLIRP networking static ipv4 to use for the VM host (as viewed from the
+# guest). Can be any ip on the 10.0.2.0/24 subnet. 10.0.2.2 is the QEMU
+# default, if none is specified.
+set vm_host_ip "10.0.2.2"
+# Set the login username
+set USER root
+
+# boot the VM with the run-vm script
+spawn [lindex $argv 0] -s -- -device e1000,netdev=n1 -netdev user,id=n1,ipv6=off,net=10.0.2.0/24,host=$vm_host_ip
+
+# getty
+# run-postinsts on a non-accelerated VM might take ~300 seconds, so increase
+# the timeout to provide plenty of time.
+set timeout 900
+expect {
+	"login: " { send "root\r" }
+	timeout {
+		send_user "TIMEOUT while waiting for getty."
```

Comment:
Should this line get a \r as well?

Reply:
Yeah. Also, in the process of triple checking my understanding to your comment above, I have figured out a better way to write these `send_user` calls. Will fix in V4.